drivers/timer/amd_tmrctr.c

```diff
@@ -0,0 +1,321 @@
+					       uint32_t reset_value)
+{
+	xlnx_tmrctr_write32(dev, counter_number, reset_value, XTC_TLR_OFFSET);
+}
+
+static inline void xlnx_tmrctr_set_options(const struct device *dev, uint8_t counter_number,
+					   uint32_t options)
+{
+	xlnx_tmrctr_write32(dev, counter_number, options, XTC_TCSR_OFFSET);
+}
+
+static int xlnx_tmrctr_start(const struct device *dev)
+{
+	struct xilinx_timer_data *data = dev->data;
+	uint32_t control_status_register;
+
+	control_status_register = xlnx_tmrctr_read32(dev, data->clockevent_offset,
+						     XTC_TCSR_OFFSET);
+	xlnx_tmrctr_write32(dev, data->clockevent_offset, XTC_CSR_LOAD_MASK, XTC_TCSR_OFFSET);
+	xlnx_tmrctr_write32(dev, data->clockevent_offset,
+			    control_status_register | XTC_CSR_ENABLE_TMR_MASK, XTC_TCSR_OFFSET);
+
+
+	control_status_register = xlnx_tmrctr_read32(dev, data->clocksource_offset,
+						     XTC_TCSR_OFFSET);
+	xlnx_tmrctr_write32(dev, data->clocksource_offset, XTC_CSR_LOAD_MASK, XTC_TCSR_OFFSET);
+	xlnx_tmrctr_write32(dev, data->clocksource_offset,
+			    control_status_register | XTC_CSR_ENABLE_TMR_MASK, XTC_TCSR_OFFSET);
+
+	return 0;
+}
+
+static int xilinx_timer_init(const struct device *dev)
+{
+	int status;
+	const struct xilinx_timer_config *config = dev->config;
+	struct xilinx_timer_data *data = dev->data;
+
+	LOG_DBG("%s: %d: Timer init at base 0x%lx, IRQ %d, clock %d, one_timer %d",
+		__func__, config->instance, config->base, config->irq, config->clock_rate,
+		config->one_timer);
+
+	/* Initialize both timers - pretty much timer reset */
+	status = xlnx_tmrctr_initialize(dev);
+	if (status != 0) {
+		return status;
+	}
+
+	if (!sys_dev && !config->one_timer) {
+		/* Doing assignment which timer is clockevent/clocksource by it's offset in IP */
+		data->clockevent_offset = 0;
+		data->clocksource_offset = TIMER_REG_OFFSET;
+
+		xlnx_tmrctr_set_reset_value(dev, data->clockevent_offset,
+					    (config->clock_rate / CONFIG_SYS_CLOCK_TICKS_PER_SEC) -
+					    1UL);
+		xlnx_tmrctr_set_options(dev, data->clockevent_offset,
+					XTC_CSR_ENABLE_INT_MASK |
+					XTC_CSR_AUTO_RELOAD_MASK |
+					XTC_CSR_DOWN_COUNT_MASK);
+
+		xlnx_tmrctr_set_options(dev, data->clocksource_offset, XTC_CSR_AUTO_RELOAD_MASK);
+
+		status = xlnx_tmrctr_start(dev);
+		if (status != 0) {
+			return status;
+		}
+
+		last_cycles = xlnx_tmrctr_read_hw_cycle_count(dev);
+
+		/* Assigning this instance as default timer */
+		sys_dev = dev;
+	}
+
+	if (config->irq_config_func != NULL) {
+		config->irq_config_func(dev);
+	}
+
+	return 0;
+}
+
+/*
+ * @brief Overwrite cycle based busy wait
+ * Implementation is derived from z_impl_k_busy_wait@kernel/timeout.c
+ *
+ * @param usec_to_wait
+ * @note Microblaze arch already implements an unaccurate, nop based
+ * no-timer-required busy wait. This routine simply overrides it with
+ * a much more accurate version.
+ */
+void arch_busy_wait(uint32_t usec_to_wait)
```

Comment:
Why do you need this?  If all you're doing is polling the same timer that implements the kernel cycle count, you might as well use the default k_busy_wait() which does exactly the same thing.  The reason this API exists is for platforms that can't achieve microsecond-scale with their timer drivers and need to use e.g. counted instruction loops or whatever.

I guess I'm gathering from the comment that this **was** true for some ancestral variant but now isn't for this driver?  In that case you just want to turn off the older implementation, not double-undo it with extra code.